services/web/src/helpers/modal.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { Modal } from 'semantic';
+
+export default function modal(Component) {
+  class ModalWrapper extends React.Component {
+    state = {
+      open: this.props.open,
+      modelKey: Date.now(),
+    };
+
+    componentDidUpdate(lastProps) {
+      const { open } = this.props;
+      if (open && open !== lastProps.open) {
+        this.setState({
+          open,
+        });
+      }
+    }
+
+    onReset = () => {
+      this.setState({ modelKey: Date.now() });
+      if (this.props.onReset) this.props.onReset();
+    };
+
+    onClose = (success) => {
+      this.setState({ open: false });
+      if (this.props.onClose) this.props.onClose(success);
+      if (this.props.onSave && success) {
```

Comment:
`onClose` is very different from `onSave`... let's discuss this one

Reply:
my thinking on that was try combine the onClose with a success argument instead of having 2 methods. 
onClose and onSave. 

a bit similar to process exit codes e.g. 0 or 1